### PR TITLE
Support for multiple files upload and drag-and-drop

### PR DESCRIPTION
**Pull request by giuband**
Added the following things:

1. Support for multiple file uploads (requires a new handy python package)
2. Support for drag and drop
3. Added a upload progress bar

All of them on the HTML5 version.

### CODE REVIEW

File: accounts/forms.py [hunk 1]
```diff
@@ -24,18 +24,21 @@
...
 from utils.spam import is_spam
 
 
-def validate_file_extension(value):
-    if not filename_has_valid_extension(str(value)):
-        raise forms.ValidationError('Uploaded file format not supported or not an audio file.')
+def validate_file_extension(audiofiles):
+    for file_ in audiofiles:
+        content_type = file_.content_type
+        if not content_type.startswith("audio") and not filename_has_valid_extension(str(file_)):
```
Comment (bdejong) [inline accounts/forms.py]: I find and/not combos always rather confusion. Do you mean:

```
if not (content_type.startswith("audio") or filename_has_valid_extension(str(file_))):
```

or rather

```
if not (content_type.startswith("audio") and filename_has_valid_extension(str(file_))):
```

File: accounts/views.py [hunk 2]
```diff
@@ -832,23 +832,25 @@ def upload_file(request):
...
             if form.is_valid():
-                if handle_uploaded_file(request.user.id, request.FILES["file"]):
-                    uploaded_file = request.FILES["file"]
-                    success = True
-                else:
-                    error = True
+                submitted_files = request.FILES.getlist('files')
+                for file_ in submitted_files:
+                    if handle_uploaded_file(request.user.id, file_):
+                        uploaded_file = file_
```
Comment (bdejong) [inline accounts/views.py]: If there are multiple uploads, uploaded_file will only contain the last one...

**Comment by bdejong** [comment]
@ffont how difficult is it right now to

- run automated tests on branches/forks in github
- get freesound to run on a local machine 😄 

@giuband COOL addition! Very minor comments... I suppose all the JS magic is just ... correct 😃

File: accounts/forms.py (hunk 1 above)
Comment (giuband) [inline accounts/forms.py]: The one I mean is the second one you propose, and you're right that is a bit confusing (actually I just realized it is wrong).

File: templates/accounts/upload.html [hunk 3]
```diff
@@ -155,11 +166,84 @@
...
+    <!--<![endif]-->
+</object>
+<p>
+<img src="{{media_url}}/images/fugue-icons/icons/exclamation.png"> <span style="color:black">Don't leave before upload finishes!</span><br/>
+Hint: you can upload multiple files at once</p>
+{% endif %}
+
+</div>
+
+<div id="describe_button_block">
```
Comment (alastair) [inline templates/accounts/upload.html]: formatting?

**Pull request merged**